aspnetcore/release-notes/aspnetcore-5.0.md

```diff
@@ -0,0 +1,464 @@
+
+### Lazy load assemblies
+
+Blazor WebAssembly app startup performance can be improved by deferring the loading of some application assemblies until they're required. For more information, see <xref:blazor/webassembly-lazy-load-assemblies>.
+
+### Updated globalization support
+
+Globalization support is available for Blazor WebAssembly based on International Components for Unicode (ICU). For more information, see <xref:blazor/globalization-localization>.
+
+## gRPC
+
+Many preformance improvements have been made in [gRPC](https://grpc.io/). For more information, see [gRPC performance improvements in .NET 5](https://devblogs.microsoft.com/aspnet/grpc-performance-improvements-in-net-5/).
+
+For more gRPC information, see <xref:grpc/index>.
+
+## SignalR
+
+SignalR Hub filters, called Hub pipelines in ASP.NET SignalR, is a feature that allows code to run before and after Hub methods are called. Running code before and after Hub methods are called is similar to how middleware has the ability to run code before and after an HTTP request. Common uses include logging, error handling, and argument validation.
+
+For more information, see [Use hub filters in ASP.NET Core SignalR](xref:signalr/hub-filters).
+
+### SignalR parallel hub invocations
+
+ASP.NET Core SignalR is now capable of handling parallel hub invocations. The default behavior can be changed to allow clients to invoke more than one hub method at a time:
+
+[!code-csharp[](~/release-notes/sample/StartupSignalRhubs.cs?name=snippet)]
+
+### Added Messagepack support in SignalR Java client
+
+A new package, [com.microsoft.signalr.messagepack](https://mvnrepository.com/artifact/com.microsoft.signalr.messagepack), adds MessagePack support to the SignalR Java client. To use the MessagePack hub protocol, add `.withHubProtocol(new MessagePackHubProtocol())` to the connection builder:
+
+```java
+HubConnection hubConnection = HubConnectionBuilder.create(
+                           "http://localhost:53353/MyHub")
+               .withHubProtocol(new MessagePackHubProtocol())
+               .build();
+```
+
+<!--
+See [Update SignalR code](xref:migration/31-to-50#signalr) for migration instructions.
+-->
+
+## Kestrel
+
+* Reloadable endpoints via configuration: Kestrel can detect changes to configuration passed to [KestrelServerOptions.Configure](xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.Configure%2A) and unbind from existing endpoints and bind to new endpoints without requiring an application restart. By default when using <xref:Microsoft.Extensions.Hosting.GenericHostBuilderExtensions.ConfigureWebHostDefaults%2A> or <xref:Microsoft.AspNetCore.WebHost.CreateDefaultBuilder%2A>, Kestrel binds to the `Kestrel` configuration subsection.
```

Comment:
I originally made this comment at https://github.com/dotnet/AspNetCore.Docs/pull/20140#discussion_r514507772, but that conversation was resolved and I don't want this to be missed.

I think think "Kestrel" should probably be left as a string literal. It's referencing the string literal passed to `IConfiguration.GetSection("Kestrel")` [here](https://github.com/dotnet/aspnetcore/blob/7e9e03b70124784b1de5564c573bd65cdaccbfcc/src/DefaultBuilder/src/WebHost.cs#L226).

We should point out that you need to pass `reloadOnChange: true` when calling `KestrelServerOptions.Configure` manually to get reloadable endpoints though.

```suggestion
* Reloadable endpoints via configuration: Kestrel can detect changes to configuration passed to [KestrelServerOptions.Configure](xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.Configure%2A) and unbind from existing endpoints and bind to new endpoints without requiring an application restart when the `reloadOnChange` parameter is `true`. By default when using <xref:Microsoft.Extensions.Hosting.GenericHostBuilderExtensions.ConfigureWebHostDefaults%2A> or <xref:Microsoft.AspNetCore.WebHost.CreateDefaultBuilder%2A>, Kestrel binds to the "Kestrel" configuration subsection with `reloadOnChange` enabled.
```